services/embedding_service.py

```
@@ -0,0 +1,43 @@
+    Generate embeddings for a list of texts using OpenAI's embedding API (v1 client).
+    Returns a list of embedding vectors (as lists of floats).
+    Retries on rate limit, with exponential backoff.
+    """
+    if not texts:
+        return []
+    client = OpenAI(api_key=get_openai_api_key(), timeout=timeout, max_retries=max_retries)
+    for attempt in range(max_retries + 1):
+        try:
+            response = client.embeddings.create(input=texts, model=model)
+            return [item.embedding for item in response.data]
+        except AuthenticationError as e:
```

Ruff (0.12.2)

Local variable `e` is assigned to but never used

Remove assignment to unused variable `e`

(F841)